framework/doc/content/syntax/Postprocessors/index.md

```diff
@@ -1,10 +1,170 @@
+postprocessor optional.
+
+!listing PostprocessorNeumannBC.C start=template end=PostprocessorNeumannBC::
+
+The actual postprocessor value must be assigned to a member variable of the class, thus in the header
+a member variable must be created, which should always be a constant reference to a
+`PostprocessorValue` type. Since this is a reference it must be initialized, this occurs in the
+source file by calling the `getPostprocessorValue` method and providing the name used in the
+validParams function. The following snippets show declaration of the reference in the header and
+the initialization of this reference in the source file.  The `_value` member variable is then
+available for use anywhere inside the object, for the case of the boundary condition it is utilized
+in the computation of the residual.
+
+!listing PostprocessorNeumannBC.h line=PostprocessorValue
+
+!listing PostprocessorNeumannBC.C start=PostprocessorNeumannBC:: end=} include-end=true
+
+## Creating a `Postprocessor` Object
+
+In general, every Postprocessor object has two methods that must be defined "execute" and
+"getValue".
+
+First, consider the execute method. This method is called by MOOSE at different time
+depending on the type of postprocessor object. Therefore, when creating a Postprocessor object
+the new object should inherit from one of the following C++ classes:
+
+- +GeneralPostprocessor+: "execute" is called once on each execution flag.
+- +NodalPostprocessor+: "execute" is called for each +node+ within the mesh on each execution flag.
+- +ElementalPostprocessor+: "execute" is called for each +element+ within the mesh on each execution
+   flag.
+- +InternalSidePostprocessor+: "execute" is called for each +side+, that is not on a boundary,
+   within the mesh on each execution flag.
+- +SidePostprocessor+: "execute" is called for each +side+, that is on a boundary, within the mesh
+   on each execution flag.
+
+The use of execution flags is discussed in the [Execute On](#execute-on) section.
+
+The getValue method is responsible for returning the value of the postprocessor object, this
+value is what is used by all objects that are coupled to the postprocessor.
+
+### Parallel Considerations
+
+When creating a postprocessor it is often necessary to perform some parallel communication
+to ensure that the value being computed is consistent across processes and threads. Three additional
+methods exists for making this process as simple as possible.
+
+- `initialize`: This is called prior to the execution of the postprocessor and should be used
+   to setup the object to be in a known state. It is important to point out that execution
+   in this context includes all calls to the execute method. For example, for a `NodalPostprocessor`
+   object the initialize method is called and then the execute method is called for all nodes.
+- `finalize`: This is called after the execution of the postprocessor and is intended to perform
```

Comment:
I think it might be worth noting that `getValue()` is sometimes used in lieu of `finalize()`. Maybe we shouldn't be doing this but the reality is that in MOOSE several Postprocessors don't define `finalize()`. What you've written is indeed correct but if a user goes poking around, they often won't see `finalize()`.

I don't know, I guess you sort of say that up above...